tests/test_md.py

```diff
@@ -354,7 +357,114 @@ def compute_reference_values():
             lamb, num_steps, np.array([], dtype=np.float64), u_interval, x_interval
         )
 
-        assert test_us.shape == (num_steps / u_interval, 0)
+        assert test_us.shape == (num_steps // u_interval, 0)
+
+    def test_local_md(self):
+        mol, _ = get_biphenyl()
+        ff = Forcefield.load_from_file("smirnoff_1_1_0_sc.py")
+
+        temperature = 300
+        dt = 1.5e-3
+        friction = 0.0
+        seed = 2022
+        radius = 1.2
+        num_steps = 500
+        x_interval = 100
+
+        unbound_potentials, sys_params, masses, coords, box = get_solvent_phase_system(mol, ff, minimize_energy=False)
+        # Select a single particle to use as the reference, will be frozen
+        local_idxs = np.array([len(coords) - 1], dtype=np.uint32)
+
+        v0 = np.zeros_like(coords)
+        bps = []
+        for p, bp in zip(sys_params, unbound_potentials):
+            bps.append(bp.bind(p).bound_impl(np.float32))
+
+        reference_values = []
+        for bp in bps:
+            reference_values.append(bp.execute(coords, box, 0.0))
+
+        intg = LangevinIntegrator(temperature, dt, friction, masses, seed)
+
+        intg_impl = intg.impl()
+
+        # If the integrator is a thermostat and temperatures don't match should fail
+        ctxt = custom_ops.Context(coords, v0, box, intg_impl, bps * 2)
+        with pytest.raises(RuntimeError) as e:
+            ctxt.local_md(np.zeros(100), local_idxs, radius=radius, temperature=200.0)
+        assert "Local MD temperature didn't match Thermostat's temperature." == str(e.value)
+
+        # Construct context with no potentials, local MD should fail.
+        ctxt = custom_ops.Context(coords, v0, box, intg_impl, [])
+        with pytest.raises(RuntimeError) as e:
+            ctxt.local_md(np.zeros(100), local_idxs, radius=radius)
+        assert "unable to find a NonbondedAllPairs potential" == str(e.value)
```

Comment:
I took this approach as it puts less burden on the user to track potentials, which I don't find trivial especially since we combined `NonbondedAllPairs` and `NonbondedPairList` in python using a `NonbondedImplWrapper`. This also seems easily testable, so don't see much risk in this approach, but subclass is doable.

Reply:
> which I don't find trivial especially since we combined NonbondedAllPairs and NonbondedPairList in python using a NonbondedImplWrapper

Ah, good point that the user may not be aware of the `NonbondedAllPairs` behind the scenes if they're using `Nonbonded`. So maybe it'd make sense for `LocalMDContext` to accept either `Nonbonded` or `NonbondedAllPairs` as a separate argument? In any case, I don't mean for this to be a blocker. Maybe we can revisit during the next deboggling week?